### PR TITLE
Removes "gatekeepers", fixes #44

### LINKED ISSUE #44: Strict mode can't really be a security solution
I do not think it is good to say, or even imply in an IETF standard that a firewall can provide any sort of proper defense against an attacker manipulating data in a protocol. The only proper solutions to that are either signing the data or the end parser doing the checks, and really only signing is a proper solution. 

I also think the text is problematic because it assumes an attacker can add data, but not modify data, so it only describes defenses against adding data.

The remedy in the text to me would be to remove the discussion of an attacker, or just mention it as one motivation with a caveat that strict mode doesn't actually provide any strong defense.

### PR DESCRIPTION
**Pull request by paulehoffman**

### CODE REVIEW

File: draft-ietf-cbor-7049bis.md [hunk 1]
```diff
@@ -2072,15 +2072,14 @@ appropriate resource management to mitigate these attacks.  (Items for
...
 
-Applications where a CBOR data item is examined by a gatekeeper
-function and later used by a different application may exhibit
-vulnerabilities when multiple interpretations of the data item are
-possible. For example, an attacker could make use of duplicate keys in
-maps and precision issues in numbers to make the gatekeeper base its
+Protocols that are used in a security
+context should be defined in such a way that multiple
+interpretations are reliably reduced to a single one. 
+For example, an on-path attacker could make use of duplicate keys in
```
Comment (cabo) [inline draft-ietf-cbor-7049bis.md]: Why does the attacker have to be on-path?  An off-path attacker could still inject traffic that has multiple interpretations.

File: draft-ietf-cbor-7049bis.md [hunk 2]
```diff
@@ -2072,15 +2072,14 @@ appropriate resource management to mitigate these attacks.  (Items for
...
 overflow vulnerabilities.)
 
-Applications where a CBOR data item is examined by a gatekeeper
-function and later used by a different application may exhibit
-vulnerabilities when multiple interpretations of the data item are
-possible. For example, an attacker could make use of duplicate keys in
-maps and precision issues in numbers to make the gatekeeper base its
+Protocols that are used in a security
+context should be defined in such a way that multiple
+interpretations are reliably reduced to a single one.
```
Comment (cabo) [inline draft-ietf-cbor-7049bis.md]: At this point "multiple interpretations" hasn't been introduced yet.  Maybe just say "multiple potential interpretations".

File: draft-ietf-cbor-7049bis.md (hunk 1 above)
Comment (paulehoffman) [reply draft-ietf-cbor-7049bis.md]: It was an example, not definitive. Feel free to remove it if you want.

File: draft-ietf-cbor-7049bis.md (hunk 2 above)
Comment (paulehoffman) [reply draft-ietf-cbor-7049bis.md]: That works too.

**Pull request merged**